### PR TITLE
[DSS-328] Upload Card - Layout Update

### PR DESCRIPTION
**Pull request by anechol**
## Description

Refactors the Upload Card layout to match with new [Figma](https://www.figma.com/file/sMbtLUHSt2vfKgKjyQ3052/branch/Hs9wohE5DRK3luSrVkWzrj/%F0%9F%A7%A9-Sage-components?type=design&node-id=2868%3A22277&mode=design&t=JKPCVczxqsfgzdcs-1) specs. Descriptions have been added to the React stories for more context. New props have been added to the React component to accommodate the refactor and match the Rails version.

**Note**: Because of one usage within `kp`, the `inputProps` prop will remain in React. Additionally, this **should not be merged** in until the `kp` version of the Upload Card has been refactored to accommodate layout changes. There will be a separate PR for this effort.

## Screenshots

|  Before  |  After  |
|--------|--------|
|||
Screen Shot 2023-08-15 at 4 09 18 PM|Screen Shot 2023-08-15 at 4 40 46 PM

Stacked Layout
Screen Shot 2023-08-15 at 4 08 30 PM

### Rails
- Navigate to [Rails component](http://localhost:4000/pages/component/upload_card?tab=preview)
- Check that layout matches Figma specs
- Check that stacked version matches Figma specs

### React
- Navigate to [React component](http://localhost:4100/?path=/docs/sage-uploadcard--default)
- Check that layout matches Figma specs
- Check that stacked version matches Figma specs

## Testing in `kajabi-products`

(**BREAKING**) Refactors the Upload Card layout to match with new Figma specs. Will affect existing instances of the Upload Card component in `kp` that utilize Sage classes for styling.

## Related

[DSS-328](https://kajabi.atlassian.net/browse/DSS-328)

[DSS-328]: https://kajabi.atlassian.net/browse/DSS-328?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/lib/sage_rails/app/views/sage_components/_sage_upload_card.html.erb [hunk 1]
```diff
@@ -1,42 +1,56 @@
...
-  <%= component.generated_html_attributes.html_safe %>
->
+<%
+  upload_card_classes = ['sage-upload-card']
+  upload_card_classes << 'sage-upload-card--stack-only' if component.stack_layout == true
+  upload_card_classes << 'sage-upload-card--selected' if component.file_selected.present?
+  upload_card_classes << 'sage-upload-card--error' if component.has_error.present?
+  upload_card_classes << component.generated_css_classes
+  upload_card_input_label = component.selection_label.present? ? component.selection_label : "Select a file"
+  upload_card_imagetypes = component.accepted_file_types.present? ? component.accepted_file_types.join(",") : "image/*"
```
Comment (ju-Skinner) [inline docs/lib/sage_rails/app/views/sage_components/_sage_upload_card.html.erb:8]: small nitpick...rename `upload_card_imagetypes` to `upload_card_image_types`

File: docs/lib/sage_rails/app/views/sage_components/_sage_upload_card.html.erb [hunk 2]
```diff
@@ -1,42 +1,56 @@
...
+    class: "sage-upload-card__preview",
+    alt: ""
+  }
+%>
+<div class="<%= upload_card_classes.join(" ") %>" <%= component.generated_html_attributes.html_safe %>>
   <div class="sage-upload-card__dropzone">
-    <input class="sage-upload-card__input" accept=".csv" multiple="multiple" type="file" autocomplete="off" />
-    <% if component.file_selected.present? %>
+    <% unless component.custom_file_input_field %>
+      <input class="sage-upload-card__input" accept="<%= upload_card_imagetypes %>" multiple="multiple" type="file" autocomplete="off" name="<%= component.name.present? ? component.name : component.id %>" id="<%= component.id %>" />
```
Comment (ju-Skinner) [inline docs/lib/sage_rails/app/views/sage_components/_sage_upload_card.html.erb:18]: should you follow the suggestion above, be sure to rename it here as well.

Comment (anechol) [reply docs/lib/sage_rails/app/views/sage_components/_sage_upload_card.html.erb]: This has been updated

**Review by ju-Skinner — APPROVED** [review]
Overall looks good. Left a few comments which do not block approval.

File: docs/app/views/examples/components/upload_card/_preview.html.erb [hunk 3]
```diff
@@ -1,42 +1,109 @@
...
+<% end %>
 
-<h3 class="t-sage-heading-6">Selected Error State</h3>
+<h3>Custom content areas</h3>
+<h4>Instructions area</h4>
+<p>The <code>sage_upload_card_instructions</code> slot provides an area for  extended instructions and custom markup.</p>
 
+<h4>Actions area</h4>
+<p>Use the <code>sage_upload_card_actions</code> slot to replace the default button and display custom components, such as dropdowns.</p>
+<p><strong>NOTE:</strong> a file input field and label are <em>included by default in the base component</em>, as seen in the examples above. When applying a custom file input with `sage_upload_card_actions`, set <code>custom_file_input_field</code> to <code>true</code> to remove these defaults.</p>
```
Comment (pixelflips) [inline docs/app/views/examples/components/upload_card/_preview.html.erb:72]: Should the sage_upload_card_actions be wrapped in `<code>` to match the others?

File: docs/app/views/examples/components/upload_card/_preview.html.erb [hunk 4]
```diff
@@ -1,42 +1,109 @@
...
+<p>Restrict accepted file types using <code>accepted_file_types</code>.</p>
 
 <%= sage_component SageUploadCard, {
   accepted_files: [
-    {name: "contacts.csv"},
+    {name: "my-image-file.jpg"},
   ],
   file_selected: true,
+  accepted_file_types: ["image/jpg"],
+  selection_preview: "https://placekitten.com/360",
```
Comment (pixelflips) [inline docs/app/views/examples/components/upload_card/_preview.html.erb:36]: 🐈

**Review by pixelflips — APPROVED** [review]
A couple of small non-blocking comments, but really great work here! Thanks so much for the documentation updates and the tests! 👍🏼

**Pull request merged**